### PR TITLE
Port to electron

### PR DESCRIPTION
**Pull request by levthedev**
This is a PR to add an Electron `main.js` and `package.json`. I was having issues with turning `main.py` into an executable so I just used `virtualenv`. To set this up, run `virtualenv -p python3 frontend_env` in the `app` folder. After creating this virtual environment, you will need to install the normal dependencies (`pip install -r requirements.txt`) and any other dependencies (I'm pretty sure Katie has listed a couple dependencies not in the requirements.txt?). Then, running `electron .` will launch the Electron app.

This also assumes you have the opentrons_sdk module locally - an easy way to do so is to run `pip install -e /path/to/opentrons_sdk/`.

### CODE REVIEW

**Comment by Kadee80** [comment]
No issues on my end.

File: app/main.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const http = require('http')
+const child_process = require('child_process')
+const electron = require('electron')
+const {app, BrowserWindow} = electron
+
+let mainWindow
+
+function startFlaskServer() {
+  child_process.exec("source frontend_env/bin/activate && python3 main.py", function(error, stdout, stderr){
```
Comment (SimplyAhmazing) [inline app/main.js]: I was thinking why not package the flask server into an exe and set up the pyinstaller stuff. Then we'll also verify nothing on that stuff is unpackagable...

**Comment by andySigler** [comment]
Just ran and it works for me, however I am having to separate windows opening.  I see in main.js `createWindow` is called both on "ready", but also 5 seconds after the flask server starts

**Comment by SimplyAhmazing** [comment]
I was thinking for the electron port we port over all the things we have for making an app -- the build scripts, travis and appveyor configs, two package json structure, pyinstaller stuff. Thoughts?

File: app/main.js (hunk 1 above)
Comment (levthedev) [inline app/main.js]: yeah that's what I was getting at, I didn't do that because I couldn't get pyinstaller to work but if it's possible we should do it.

**Comment by levthedev** [comment]
@SimplyAhmazing I agree, I just figured this is an easy first step. I don't wanna do it all at once.

**Comment by andySigler** [comment]
@SimplyAhmazing @levthedev +1

**Comment by levthedev** [comment]
I'm actually having an issue locally with my virtualenv so I'm not gonna merge this in until I fix that. Also need to look into what @andySigler is saying - I don't get 2 windows but I will investigate.

**Comment by levthedev** [comment]
@andySigler looks like your issue occurs when you haven't installed the extra requirements in the README - for me, it was gevent that was causing it. I added the extra requirements to requirements.txt so hopefully this shouldn't be an issue anymore.

**Pull request merged**